src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs

```diff
@@ -97,15 +99,32 @@ IConnectionFactory CreateConnectionFactory(IServiceProvider sp)
             return factory;
         }
 
+        const string resilienceKey = "Aspire.RabbitMQ.Client.AspireRabbitMQExtensions.CreateConnection";
+        builder.Services.AddResiliencePipeline(resilienceKey, builder =>
```

Comment:
Do we really need to use DI for this? Is it not possible to just build a strategy object inline in the `CreateConnection` method? It just feels like unnecessary indirection.

Reply:
Creating the resilience pipeline directly would be a natural conversion of the current code (and we're not using DI for testability so there's probably no point in doing it).